pilot/pkg/networking/core/v1alpha3/listener_test.go

```diff
@@ -1403,6 +1404,121 @@ func TestHttpProxyListener(t *testing.T) {
 	}
 }
 
+func TestHttpProxyListener_CustomTags(t *testing.T) {
+	var customTagsTest = []struct {
```

Comment:
Add a test case for router and sidecar.

Reply:
Added.

@nrjpoddar For gateways, do we want to add support for literals and env and only disregard request headers from external clients?

Right now it does not respect any custom_tag for gateways.